pkg/common/cns-lib/vsphere/pbm.go

```diff
@@ -82,6 +82,11 @@ func (vc *VirtualCenter) DisconnectPbm(ctx context.Context) error {
 // GetStoragePolicyIDByName gets storage policy ID by name.
 func (vc *VirtualCenter) GetStoragePolicyIDByName(ctx context.Context, storagePolicyName string) (string, error) {
 	log := logger.GetLogger(ctx)
+	err := vc.ConnectPbm(ctx)
```

Comment:
Why can't this be part of VirtualCenter methods? This way the callers need not worry about making sure the connection is valid before invoking any vSphere API.

Reply:
When VC client is created, we do not create CNS Client and PBM Client. They are getting created on-demand when required. 

`GetStoragePolicyIDByName()` was called multiple places and everywhere we were calling `err := vc.ConnectPbm(ctx)`. so moved this inside `GetStoragePolicyIDByName`. so the caller does not need to explictly call ConnectPbm before calling `GetStoragePolicyIDByName` method.